lib/Sentry.d.ts

```diff
@@ -0,0 +1,96 @@
+// Type definitions for react-native-sentry
+// Project: https://sentry.io
+// Definitions by: Daniel Griesser <https://github.com/hazat>
+// Definitions: https://github.com/getsentry/react-native-sentry
+// TypeScript Version: 2.3
+
+type SentryBreadcrumbType = "navigation" | "http";
+
+interface SentryBreadcrumb {
+    message?: string;
+    category?: string;
+    level?: SentrySeverity;
+    data?: object;
+    type?: SentryBreadcrumbType;
+}
+
+export enum SentrySeverity {
+    Fatal = "fatal",
+    Error = "error",
+    Warning = "warning",
+    Info = "info",
+    Debug = "debug",
+    Critical = "critical"
+}
+
+export enum SentryLog {
+    None = 0,
+    Error = 1,
+    Debug = 2,
+    Verbose = 3
+}
+
+interface SentryOptions {
+    logLevel?: SentryLog;
+    instrument?: boolean;
+    disableNativeIntegration?: boolean;
+    ignoreModulesExclude?: [string];
+    ignoreModulesInclude?: [string];
+}
+
+export default Sentry;
+
+export class Sentry {
+    install(): void;
+
+    static config(dsn: string, options?: SentryOptions): Sentry;
+
+    static isNativeClientAvailable(): boolean;
+
+    static crash(): void;
+
+    static nativeCrash(): void;
+
+    static setEventSentSuccessfully(callback: Function): void;
+
+    static setDataCallback(callback: Function): void;
+
+    static setUserContext(user: {
+        id?: string;
+        username?: string;
+        email?: string;
+        extra?: object;
+    }): void;
+
+    static setTagsContext(tags: Object): void;
+
+    static setExtraContext(extra: Object): void;
+
+    static captureMessage(message: string, options?: object): void;
+
+    static captureException(ex: Error, options?: object): void;
+
+    static captureBreadcrumb(breadcrumb: SentryBreadcrumb): void;
+
+    static clearContext(): void;
+
+    static context(func: Function, ...args: any[]): void;
```

Comment:
This and some other functions actually call through to `raven-js` and return its return value (which happens to be `void` in these cases). Once [typescript#6606](https://github.com/Microsoft/TypeScript/issues/6606) is closed, we could import raven-js' typescript definition and use the `typeof` operator to infer the return type